hdr/sqlite_modern_cpp.h

```diff
@@ -85,11 +85,19 @@ namespace sqlite {
 			return ++_inx;
 		}
 
-		sqlite3_stmt* _prepare(const std::u16string& sql) {
-			return _prepare(utility::utf16_to_utf8(sql));
+		sqlite3_stmt* _prepare(const U16STR_REF& sql) {
+			//return _prepare(utility::utf16_to_utf8(sql));
+			int hresult;
+			sqlite3_stmt* tmp = nullptr;
+			const void *remaining;
+			hresult = sqlite3_prepare16_v2(_db.get(), sql.data(), -1, &tmp, &remaining);
+			if (hresult != SQLITE_OK) errors::throw_sqlite_error(hresult, utility::utf16_to_utf8(sql.data()));
+			if (!std::all_of(static_cast<const char16_t*>(remaining), sql.data() + sql.size(), [](char16_t ch) {return std::isspace(ch); }))
+				throw errors::more_statements("Multiple semicolon separated statements are unsupported", utility::utf16_to_utf8(sql.data()));
+			return tmp;
```

Comment:
Is there an advantage in not reusing the UTF-8 version?
I see some smaller problems here:

- The SQL statement in the exception is no longer the same statement given to SQLite.
In edge cases `utf16_to_utf8` might behave differently than SQLite which can make debugging hard.
- Less usage of `utf16_to_utf8` in main code paths might lead to problems in this function to go unnoticed which again makes debugging a nightmare if they only surface at error conditions.
- It is marginally slower because in case of an error the conversion happens twice.
- And of course we introduce some redundant code.